share/change.py

```diff
@@ -193,7 +218,12 @@ def __init__(self, graph):
         self.__graph = graph
         self.__visted = set()
         self.__visiting = set()
-        self.__nodes = list(graph.nodes)
+
+        # visit identifiers first
+        self.__nodes = sorted(
```

Comment:
If I'm reading it right, the topological sort ensures that any through-node is visited after the nodes it connects, but doesn't guarantee anything about the relative order of identifiers, works, people, etc. Identifiers are topologically equivalent (as it were) to the objects they identify, but need to be visited first. Sorting the list seemed simplest, but it'll be a special case no matter what.